### ISSUE TITLE
[BUG] Poor Performance on Rog Ally Z1E

### ISSUE THREAD

**Issue by mufaaf** [streetpea/chiaki-ng#395]
Hey folks. This is a very recent issue.

Chiaki's performance on the ROG Ally Z1E has degraded tremendously. It's the same with 1.8.1 and other versions. There is constant packet loss, glitches, stuttering and artifacting. I have tried playing around the bitrate, codecs, encoders, etc but nothing seems to work. However, it works flawlessly on the MacOS.

PSPlay on the Ally works just fine, over the same setup and network connections (albeit I have only tried the demo version but it should be the same on the fully licensed version).

I am not sure what went wrong, and how I can go about fixing it. I have tried resetting the WiFi drivers, the system is upto date, I have tried lower and higher TDP levels, but nothing seems to help.

**Comment by streetpea** [comment]
Can you post a log from a session using your Rog Ally?

**Comment by mufaaf** [comment]
> @mufaaf can you post a log from a session using your Rog Ally?

Hey, I have attached the most recent log file here.
[chiaki_session_2024-08-24_00-11-43-342342.log](https://github.com/user-attachments/files/16733465/chiaki_session_2024-08-24_00-11-43-342342.log)

I hope you can devise some solution. I just tried PSPlay and it works flawlessly but with small stutters that is inherent to streaming but there is greater latency compared to Chiaki. On MacOS, Chiaki works flawlessly.

**Comment by streetpea** [comment]
@mufaaf I can't reproduce on my Windows 10 PC. It seems to run flawlessly...my ping avg when connecting is 1.667ms while yours is 5.820 ms ... I also don't see any of the drops you're getting. I don't have a Windows 11 device to test if maybe there is something going wrong on there. Can you try with a Windows 10 device or another Windows 11 device and try to reproduce?

**Comment by mufaaf** [comment]
> @mufaaf I can't reproduce on my Windows 10 PC. It seems to run flawlessly using 1.8.1...my ping avg when connecting is 1.667ms while yours is 5.820 ms ... I also don't see any of the drops you're getting. I don't have a Windows 11 device to test if maybe there is something going wrong on there. Can you try with a Windows 10 device or another Windows 11 device and try to reproduce?

Just tried on my Windows 10 desktop PC and had no issues there. Maybe a powerful PC is able to handle the streaming just fine. Even Ally the was but suddenly it's bugging out. Is there a way to completely wipe C4D off the PC and start a fresh? Maybe that could help?

**Comment by streetpea** [comment]
I tested using an amd Radeon pro 455 on my 2016 macbook Pro (under bootcamp) which is weaker than the gpu in the rog ally so theoretically it shouldn’t be an issue to run it on the ally. Having side that it is windows 10 not 11 so maybe an issue there. You can manually delete the settings in the windows registry editor if you open it and search for Chiaki if you want to start from scratch but I honestly doubt that will help.

**Comment by streetpea** [comment]
You can try changing the render preset to fast and see if that helps at all.

**Comment by mufaaf** [comment]
> You can try changing the render preset to fast and see if that helps at all. That should work fine on resource constrained systems but honestly it runs easily on my Steam Deck as well on high quality and that is weaker than the Ally as well and it runs for 6-8 hours at that so isn't particularly using a lot of power ... for comparison triple a games last 1-2 hours.

Yeah, my thoughts exactly. I have tried the Fast preset but the result is the same. Even 10k bitrate, but same.

Not sure what the hang up is when PSPlay and even Moonlight run flawlessly (or at least as expected).

It did use to run flawlessly but then something changed somewhere and I can't figure it out. I guess I have no option other than to purchase PSPlay, but I dislike it's greater latency compared to Chiaki. Sigh.

**Comment by streetpea** [comment]
It looks like you’re getting a lot of corrupted frames. @nowrep do you have an idea of what could be going on here. It seems somewhat surprising that other streaming apps are working given it seems like Chiaki-ng is constantly getting corrupted frames.

**Comment by nowrep** [comment]
I'm not sure, this just looks like network issue.

**Comment by mufaaf** [comment]
> It looks like you’re getting a lot of corrupted frames. @nowrep do you have an idea of what could be going on here? It seems somewhat surprising that other streaming apps are working given it seems like Chiaki-ng is constantly getting corrupted frames.

So I went back to 1.6.3, and what do ya know, no issues whatsoever! No packet drops, no frame drops, and the stream is BUUTERY smooth actual 60fps. I am not sure why this version works and the latest doesn't, but at least I have a solution!

**Comment by mufaaf** [comment]
> > It looks like you’re getting a lot of corrupted frames. @nowrep do you have an idea of what could be going on here? It seems somewhat surprising that other streaming apps are working given it seems like Chiaki-ng is constantly getting corrupted frames.
> 
> So I went back to 1.6.3, and what do ya know, no issues whatsoever! No packet drops, no frame drops, and the stream is BUUTERY smooth actual 60fps. I am not sure why this version works and the latest doesn't, but at least I have a solution!

Well, that was extremely short-lived. The dropped frames and packet losses return on the second sessions, but it happens far less than 1.8.1. I am so confused and frustrated with this. When it works, it's glorious but super smooth 60fps, but then it sputters and stutters, just like the Portal and I hate it.

I am not even sure what I did differently between the two sessions. I simply logged in and out.

**Comment by mufaaf** [comment]
I reset the Ally entirely and ran C4D on completely stock settings, and the problem still persists. I have tried everything. 1.6.3 is the best version so far. Maybe I will try going down even further and see. I remember 1.6.1 offering the smoothest streaming so I will give that a shot.

**Comment by streetpea** [comment]
@mufaaf have you tried 720p with a bitrate of 10000?

**Comment by mufaaf** [comment]
> @mufaaf have you tried 720p with a bitrate of 10000? Also make sure you have metered connection off for wifi https://support.microsoft.com/en-us/windows/metered-connections-in-windows-7b33928f-a144-b265-97b6-f2e95a87c408#:~:text=Select%20Start%20%3E%20Settings%20%3E%20Network%20%26,on%20Set%20as%20metered%20connection.

Tried 720p, 10K even before I saw your comment, and nope, still the same. Metered connection was already off. :(

C4D works perfectly on my Windows desktop and MBA. But just the fucking Ally is causing problems. Even a goddman reset didn't fix whatever the issue was.

**Comment by mufaaf** [comment]
As a last ditch effort, I installed BazziteOS on the Ally to see if C4D would perform better, and my hopes were quickly dashed. It performs the same, or maybe slightly better. But the constant packet loss and frame drops continue to happen. Tried all manner of settings but the same thing, unfortunately. It seems like a hardware issue on the Ally or something else that's not sitting right with C4D.

Massive bummer. Also, the rumble doesn't work on Bazzite but that's the least of my problems, lol.

**Comment by antnard1** [comment]
I'm using chiaki-ng in bazzite and loving it (same issue with no rumble though).  There is some wifi bug that affects both the steam deck and bazzite running on the ally which causes it to scan for optimal access points every few minutes, which causes it to have massive packet loss while streaming.  in desktop mode, go to wifi settings for your configured wifi connection, Mode: infrastructure, BSSID: select the mac address of your preferred AP/SSID.  I have several APs each broadcasting several bands, and picking the mac address for my 6GHz network from my most-central AP in my house completely cleared up the packet loss issue from when it was doing interval scans.  Of course YMMV but give it a shot.

**Comment by mufaaf** [comment]
> I'm using chiaki-ng in bazzite and loving it (same issue with no rumble though). There is some wifi bug that affects both the steam deck and bazzite running on the ally which causes it to scan for optimal access points every few minutes, which causes it to have massive packet loss while streaming. in desktop mode, go to wifi settings for your configured wifi connection, Mode: infrastructure, BSSID: select the mac address of your preferred AP/SSID. I have several APs each broadcasting several bands, and picking the mac address for my 6GHz network from my most-central AP in my house completely cleared up the packet loss issue from when it was doing interval scans. Of course YMMV but give it a shot.

I saw a similar comment on Reddit and did the same but no luck. Chiaki still runs poorly no matter what. 

I bought PSPlay today, and on Windows, it runs marvelously smooth with very minimal stutter and hiccups here and there.

Not sure why one software, on two separate OS, does not run correctly, while the other, both designed to do the same thing, runs just perfectly fine.

**Comment by streetpea** [comment]
Hard to say … they both use libplacebo as the renderer and psplay actually improved its error handling for stream corruption based on Chiaki.

**Comment by mufaaf** [comment]
> Hard to say … they both use libplacebo as the renderer and psplay actually improved its error handling for stream corruption based on chiaki-ng. Also Chiaki-ng runs perfectly smooth on both my windows device and Steam Deck OLED as well as Steam Deck LCD but I have optimized my internet connection. Perhaps it’s doing something else to handle errors. In both cases you should be getting errors bc the errors that are coming up in your logs are at the windows network level. It just seems that for whatever reason Chiaki-ng is more sensitive to the particular errors you are getting.

Yeah, it does seem like Chiaki can't handle my network, even though mine is optimized in the best way I can do. I have a Mesh setup, but both the PS5 and the Ally connect to the main AP. I have no issues with any other streaming service, including YT, Stremio, Prime, etc and even Moonlight and PSPlay. But Chiaki stumbles for some reason. Baffling.

Chiaki works great on my desktop and on MacOS, but the Ally is a different beast.

Also, love the fact that you and PSPlay dev work together to better each other's softwares.

**Comment by streetpea** [comment]
For me they both work similarly on Steam Deck but Chiaki-ng seems to have a slightly better framerate / frame consistency. I’d just say if psplay works better for you use that.

**Comment by streetpea** [comment]
@mufaaf you could try changing the packet loss max % to 100% and see if that improves anything for you.

**Comment by mufaaf** [comment]
> @mufaaf you could also try changing the packet loss max % to 100% and see if that improves anything for you that should favor consistency over resolution.

Didn't work. Same packet loss and frame drops :(. Ahh well, at least PSPlay works!
